src/docfx/build/schema/BuildSchemaDocument.cs

```diff
@@ -29,7 +29,8 @@ internal static class BuildSchemaDocument
                 throw Errors.SchemaNotFound(schema).ToException();
             }
 
-            var content = token.ToObject(schemaType);
+            var (mismatchingErrors, content) = token.ToObjectAndValidateMismatchingFieldType(schemaType);
```

Comment:
>ToObjectAndValidateMismatchingFieldType [](start = 53, length = 39)

this is too long, consider just `ToObject` an make it an non-extension method, we can just call `JsonUtility.ToObject(token)` instead